packages/api/cms-api/src/auth/util/auth-service.interface.ts

```diff
@@ -2,6 +2,15 @@ import { type Request } from "express";
 
 import { type User } from "../../user-permissions/interfaces/user";
 
+export const SKIP_AUTH_SERVICE = "skip-auth-service" as const;
+
+export type AuthenticateUserResult =
+    | { systemUser: string }
+    | { userId: string }
+    | { authenticationError: string }
+    | { user: User }
+    | typeof SKIP_AUTH_SERVICE;
+
 export type AuthServiceInterface = {
-    authenticateUser: (request: Request) => Promise<User | string | undefined> | User | string | undefined;
+    authenticateUser: (request: Request) => Promise<AuthenticateUserResult> | AuthenticateUserResult;
```

Comment:
@fraxachun Did something go wrong during the rebase? These are the changes that were already added int https://github.com/vivid-planet/comet/pull/3516/files